processing/src/main/java/io/druid/segment/filter/LikeFilter.java

```diff
@@ -140,4 +103,107 @@ public boolean supportsBitmapIndex(BitmapIndexSelector selector)
   {
     return selector.getBitmapIndex(dimension) != null;
   }
+
+  @Override
+  public double estimateSelectivity(BitmapIndexSelector selector, long totalNumRows)
+  {
+    if (extractionFn == null && likeMatcher.getSuffixMatch() == LikeDimFilter.LikeMatcher.SuffixMatch.MATCH_EMPTY) {
```

Comment:
Maybe have `getBitmapIterator` take some higher level args and even handle the fallback vs. optimized case internally. In the fallback case it could return an `Iterators.singletonIterator` of the one bitmap. Then `getBitmapIndex` unions them and `estimateSelectivity` just adds up their sizes.

The helpers in Filters.java, as well as specific filters like BoundFilter and SpatialFilter, maybe others too, could use a similar strategy to reduce duplicated code.